lib/ansible/config/manager.py

```diff
@@ -390,7 +390,11 @@ def _loop_entries(self, container, entry_list):
         origin = None
         for entry in entry_list:
             name = entry.get('name')
-            temp_value = container.get(name, None)
+            try:
+                temp_value = container.get(name, None)
+            except UnicodeEncodeError:
+                self.WARNINGS.add('value for config entry {0} contains invalid characters, ignoring...'.format(to_native(name)))
```

Comment:
This is not correct.... and it also leads me to see that the code which prints self.WARNINGS is not correct :-(  This is going to be a short rabbit hole but probably not appropriate for 2.9.0.  I notice that @sivel's code does not emit a warning.  I think there was a reason and simply passes over this entry.  I think there was a reason (that's what happens on Python3, perhaps?)

So if sivel can confirm that, then maybe that's what we should do here.

Reply:
The reason this is not correct, is that it should set text strings, rather than native strings.  When I looked at the code that outputs this, I found that one pathway needs text strings and the other pathway needs native strings.  That all needs to be fixed so that we input text strings into self.WARNINGS and then the output code converts it to byte strings when appropriate (either inside of display as done now, or just before outputting to sys.stderr on python2 or outputting to sys.stderr.buffer on python3).

Reply:
I wish I could remember. I only remember that silently ignoring was what we agreed at the time. I'd have to go digging to see if py3 does that.

Reply:
Python3 looks like it will raise an exception on `__getitem__`.  So that wasn't it.  I couldn't find context, or jog my memory.

Reply:
It looks like Python3 doesn't raise an error.  It surrogateescapes:
```
[toshio@el7-test ~][PROD]$ TESTING='café' LC_ALL=C  python3.6 test.py
'caf\udcc3\udca9'
[toshio@el7-test ~][PROD]$ cat test.py
import os
a = os.environ.get('TESTING')
print(repr(a))
```

Reply:
Okay, Looked waaaaay back in my slack history and we decided a warning wasn't necessary because of exactly this rabbit hole.

So if you do want to emit a warning and, as talked about on slack, we're only dealing with ascii values for name, then this should just be using text and then it would be fine:
```suggestion
                self.WARNINGS.add(u'Ignoring the value for config entry {0} as it contains non-utf8 characters'.format(to_text(name)))
```